src/mock_sim.h

```diff
@@ -0,0 +1,191 @@
+/// framework:
+///
+/// @code
+///
+///  cyclus::QueryResult qr = sim.db().Query("Transactions", NULL);
+///  int n_trans = qr.rows.size();
+///  EXPECT_EQ(10, n_trans) << "expected 10 transactions, got " << n_trans;
+///
+/// @endcode
+class MockSim {
+ public:
+
+  /// Creates and initializes a new mock simulation environment to test the
+  /// archetype identified by spec.  config should contain the
+  /// archetype-specific xml snippet excluding the wrapping "<config>" and
+  /// "<[AgentName]>" tags.  duration is the length of the simulation in time
+  /// steps.
+  MockSim(AgentSpec spec, std::string config, int duration);
+
+  /// AddSource adds a source facility that can offer/provide material to the
+  /// archetype being tested.  commod specifies the commodity the source will
+  /// offer on. AddSource can be called multiple times to generate many sources
+  /// for the simulation.  The returned MockAgent object has several functions
+  /// that can be called to configure the source's behavior further.  Don't
+  /// forget to call the MockAgent object's "Finalize" function when you are
+  /// done configuring it.  MockAgent's functions support chaining:
+  ///
+  /// @code
+  ///
+  /// cyclus::MockSim sim(...);
+  /// sim.AddSource("fresh_mox")
+  ///    .start(7).lifetime(3).recipe("mox_fuel")
+  ///    .Finalize();
+  ///
+  /// @endcode
+  MockAgent AddSource(std::string commod);
+
+  /// AddSink adds a sink facility that can request+receive material from the
+  /// archetype being tested.  commod specifies the commodity the sink will
+  /// request on. AddSink can be called multiple times to generate many sinks
+  /// for the simulation.  The returned MockAgent object has several functions
+  /// that can be called to configure the sink's behavior further.  Don't
+  /// forget to call the MockAgent object's "Finalize" function when you are
+  /// done configuring it.  MockAgent's functions support chaining:
+  ///
+  /// @code
+  ///
+  /// cyclus::MockSim sim(...);
+  /// sim.AddSink("spent_mox")
+  ///    .start(7).lifetime(3).recipe("spent_fresh_mox")
+  ///    .Finalize();
+  ///
+  /// @endcode
+  MockAgent AddSink(std::string commod);
+
+  /// AddRecipe adds a recipe to the mock simulation environment (i.e. to the
+  /// simulation context).  Any recipes that your archetype expects to find in
+  /// the simulation context must be added this way; if the xml configuration
+  /// snippet for the archetype being tested contains a recipe name, add it
+  /// with this function.
+  void AddRecipe(std::string name, Composition::Ptr c);
+
+  /// Run the simulation.  This can only be called once.  After the simulation
+  /// has been run, this MockSim object CANNOT be reused to run other
+  /// simulations.
+  void Run();
+
+  /// Returns the underlying in-memory database containing results for
+  /// the simulation.  Run must be called before the database will contain
+  /// anything.
+  SqliteBack& db();
```

Comment:
Since this is just for in-memory querying of test/mock sims, I figured it was not necessary to generalize.  The sqlitedb is entirely in-memory.  And this also gives the tester the ability to run full blown sql queries should they choose.